include/ignition/gui/qml/PlottingInterface.qml

```diff
@@ -0,0 +1,538 @@
+      charts[idIncrementor] = chartObject;
+      // Signals and Slots
+      chartObject.subscribe.connect(main.onSubscribe);
+      chartObject.unSubscribe.connect(main.onUnSubscribe);
+      chartObject.componentSubscribe.connect(main.onComponentSubscribe);
+      chartObject.componentUnSubscribe.connect(main.onComponentUnSubscribe);
+      chartObject.clicked.connect(main.onClicked);
+  }
+
+  function reallocateChart(_chart)
+  {
+      // new attributes to fit in the horizontal layout
+      _chart.width = 200;
+      _chart.height = Qt.binding( function() {return rowCharts.height * 0.9} );
+      _chart.y = Qt.binding( function() {return (rowCharts.height - _chart.height)/2} );
+      // enable that to enable the some UI features in the chart
+      _chart.multiChartsMode = true;
+      _chart.fillPlotInOrOut();
+
+      // change the layout from the vertical layout to the top horizontal one
+      _chart.parent = rowChartsLayout;
+  }
+
+  function onSubscribe(_Id, _topic, _path)
+  {
+      PlottingIface.subscribe(_Id, _topic, _path);
+  }
+  function onUnSubscribe(_Id, _topic, _path)
+  {
+      PlottingIface.unsubscribe(_Id, _topic, _path);
+  }
+  function onComponentSubscribe(entity, typeId, type, attribute, Id)
+  {
+      PlottingIface.onComponentSubscribe(entity, typeId, type, attribute, Id);
+  }
+  function onComponentUnSubscribe(entity, typeId, attribute, Id)
+  {
+      PlottingIface.onComponentUnSubscribe(entity, typeId, attribute, Id);
+  }
+  function onClicked(Id)
+  {
+      // if many charts mode & the selected chart is in the horizontal layout
+      if (multiChartsMode && charts[Id].multiChartsMode)
+      {
+          // ======= main chart =========
+          // change the main charts properties to fit in the horizontal layout
+          charts[mainChartID].width = 200;
+          charts[mainChartID].height = Qt.binding( function() {return rowCharts.height * 0.9} );
+          charts[mainChartID].y = (rowCharts.height - charts[mainChartID].height)/2;
+          charts[mainChartID].multiChartsMode = true;
+          charts[mainChartID].fillPlotInOrOut();
+
+          // swap the main chart with the position of the clicked chart
+          charts[mainChartID].parent = rowChartsLayout;
+
+          // ======= swapped chart =======
+          charts[Id].parent = chartsLayout;
+          charts[Id].x = charts[mainChartID].x;
+          charts[Id].y = 0;
+          charts[Id].height = Qt.binding( function() {return chartsLayout.height / chartsLayout.heightFactor});
+          charts[Id].width = Qt.binding( function() {return chartsLayout.width});
+          charts[Id].multiChartsMode = false;
+          charts[Id].fillPlotInOrOut();
+          charts[Id].setChartOpacity(1);
+
+          mainChartID = Id;
+      }
+  }
+
+  Connections {
+      target: PlottingIface
+      onPlot : function (_chart, _fieldID, _x, _y)
+      {
+          charts[_chart].appendPoint(_fieldID, _x, _y);
+      }
```

Comment:
I traced the problem that we're having in seeing the plotted points to the way these functions are specified. Our version of Qt/QML doesn't seem to support this format: `slotName: function(params){}`.

Instead, this worked for me and I was able to see plotted points:

```diff
diff --git a/include/ignition/gui/qml/PlottingInterface.qml b/include/ignition/gui/qml/PlottingInterface.qml
index 9b88bf7..de776e5 100644
--- a/include/ignition/gui/qml/PlottingInterface.qml
+++ b/include/ignition/gui/qml/PlottingInterface.qml
@@ -230,12 +230,14 @@ Rectangle
       }
   }
 
+   function handlePlot(_chart, _fieldID, _x, _y)
+   {
+     charts[_chart].appendPoint(_fieldID, _x, _y);
+   }
+
   Connections {
       target: PlottingIface
-      onPlot : function (_chart, _fieldID, _x, _y)
-      {
-          charts[_chart].appendPoint(_fieldID, _x, _y);
-      }
+      onPlot: handlePlot(_chart, _fieldID, _x, _y)
       onMoveChart :function() {
           Object.keys(charts).forEach(function(key) {
               charts[key].moveChart();
```

Can you change all connections to use this style?